src/ucp/core/ucp_am.c

```diff
@@ -579,13 +665,28 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_send_nb,
     return ret;
 }
 
-UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_send_nbx,
-                 (ep, id, header, header_length, buffer, count, param),
-                 ucp_ep_h ep, unsigned id, const void *header,
-                 size_t header_length, const void *buffer,
-                 size_t count, const ucp_request_param_t *param)
+ucs_status_ptr_t ucp_am_send_nb(ucp_ep_h ep, uint16_t id, const void *payload,
+                                size_t count, ucp_datatype_t datatype,
+                                ucp_send_callback_t cb, unsigned flags)
 {
-    return UCS_STATUS_PTR(UCS_ERR_NOT_IMPLEMENTED);
+    ucp_request_param_t params = {
+        .op_attr_mask = UCP_OP_ATTR_FIELD_DATATYPE |
+                        UCP_OP_ATTR_FIELD_CALLBACK |
+                        UCP_OP_ATTR_FIELD_FLAGS,
+        .flags        = flags,
+        .cb.send      = (ucp_send_nbx_callback_t)cb,
+        .datatype     = datatype
+    };
+
+    return ucp_am_send_common(ep, id, NULL, 0, payload, count, &params, 0);
```

Comment:
can we just have ucp_am_send_nbx and ucp_am_send_nb so that ucp_am_send_nb will use ucp_am_send_nbx()?
e.,g avoid ucp_am_send_common
i guess it would mean remove NBX flag on the wire and the receiver would fail if the header size is non-empty

Reply:
new API can send zero header as well. I can do reverse flag, `NB_API` or `OLD_API`, and then `ucp_am_send_nb` will set this flag and call `ucp_am_send_nbx`

Reply:
question is can we avoid the NBX flag on the wire altogether? can simplify code and reduce some checks